sdk/core/azure-core/inc/azure/core/uuid.hpp

```diff
@@ -44,71 +37,12 @@ namespace Azure { namespace Core {
      * @brief Gets Uuid as a string.
      * @details A string is in canonical format (4-2-2-2-6 lowercase hex and dashes only).
      */
-    std::string ToString()
-    {
-      // Guid is 36 characters
-      //  Add one byte for the \0
-      char s[37];
-
-      std::snprintf(
-          s,
-          sizeof(s),
-          "%2.2x%2.2x%2.2x%2.2x-%2.2x%2.2x-%2.2x%2.2x-%2.2x%2.2x-%2.2x%2.2x%2.2x%2.2x%2.2x%2.2x",
-          m_uuid[0],
-          m_uuid[1],
-          m_uuid[2],
-          m_uuid[3],
-          m_uuid[4],
-          m_uuid[5],
-          m_uuid[6],
-          m_uuid[7],
-          m_uuid[8],
-          m_uuid[9],
-          m_uuid[10],
-          m_uuid[11],
-          m_uuid[12],
-          m_uuid[13],
-          m_uuid[14],
-          m_uuid[15]);
-
-      return std::string(s);
-    }
+    std::string ToString();
 
     /**
      * @brief Creates a new random UUID.
      *
      */
-    static Uuid CreateUuid()
-    {
-      uint8_t uuid[UuidSize] = {};
-
-#if defined(AZ_PLATFORM_WINDOWS)
-      std::random_device rd;
-
-      for (size_t i = 0; i < UuidSize; i += 4)
-      {
-        const uint32_t x = rd();
-        std::memcpy(uuid + i, &x, 4);
-      }
-#elif defined(AZ_PLATFORM_POSIX)
-      // This static cast is safe since we know Uuid size, which is a const, will always fit an int.
-      int ret = RAND_bytes(uuid, static_cast<int>(UuidSize));
```

Comment:
I was thinking of encapsulating this specific function call to a non-inline private function, but making the entire functions non-inlineable probably works too. Try to get @RickWinter to review, he wrote this.

Reply:
I don't think there's any difference. After moving the entire function body to `.cpp` file, the header file now looks more concise.

Reply:
I can't disagree, it is only the perf that could be a bit different, although I don't believe it makes a difference, in this place.

Reply:
If we only keep the class declaration here, I think we should move `#include <cstring>` from the header to the source.

Reply:
We still need `<cstring>` because we're using `std::memcpy` in the private constructor.

Reply:
Ah, I see it now, thanks for clarification! GitHub hid that line, and I didn't see it. Sure then!